spec/requests/groups/membership_spec.rb

```diff
@@ -1,117 +1,132 @@
-# frozen_string_literal: true
-
-RSpec.describe Groups::MembershipsController, type: :controller do
-  include StubCurrentUserHelper
-
-  before do
-    create_current_user
-  end
-
-  describe '#create' do
-    context 'new group member for existing group' do
-      let!(:leader_user) { create :user2 }
-      let!(:group_leader) { create :group_member, user_id: leader_user.id, leader: true }
-      let!(:group) { group_leader.group }
```

Comment:
There's a handful of `let` statements declared throughout the controller spec. I like let when there's a few it blocks or context blocks that uses the same defined variable, but for one it block, I think it's harder to read and can be an anti pattern. Also with the behavior of being threadsafe or not based on config as mentioend in [the rubydoc for rspec-core](https://www.rubydoc.info/github/rspec/rspec-core/RSpec%2FCore%2FMemoizedHelpers%2FClassMethods%3Alet), I feel like we should use it when needed. 

I've seen some flakey tests from a few disorganized `let` or ambigious tests.

I personally avoid using `let!` with the exclamation point unless I have to since that will always be evaluated.

---

I'm open to any opinions or counterexamples. :)

Reply:
Thanks for sharing this insight! That makes a lot of sense!